### PR TITLE
Implement 6 va-icons on landing page body

### PR DESCRIPTION
**Pull request by dcloud**
## Summary

- Use `va-icon` component, and update icons on NavCards
- Add `iconClasses` prop that allows us to tweak the visual alignment of the _Payments_ dollar sign
- Edited `MedicalRecordsCard.jsx`, which is a custom card that is behind the `mhvTransitionalMedicalRecordsLandingPage` feature flag.

## Related issue(s)

- department-of-veterans-affairs/va.gov-team/issues/83788

## Testing done

- Visual inspection of icons

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | icons-before-2024-05-31-at-16-32-59-fullpage | icons-after-2024-05-31-at-16-52-05-fullpage |
| Desktop | icons-before-2024-05-31-at-16-33-33-fullpage | icons-after-2024-05-31-at-16-51-59-fullpage |

## What areas of the site does it impact?

`/my-health`, the MHV Landing Page

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by dcloud** [comment]
Drafted this, though the `pill` icon in the DS has [a rendering bug](https://github.com/department-of-veterans-affairs/vets-website/pull/30009), so will hold off on requesting review until that's fixed

**Comment by dcloud** [comment]
Pill rendering bug was fixed, hence the package.json and yarn.lock changes. 

CI is flaking out with some weird error that results in a timeout:

```
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)

DevTools listening on ws://127.0.0.1:46253/devtools/browser/2ef6dd1d-6088-4a9b-91b6-4edec63a0789
[1081:0603/1601[31](https://github.com/department-of-veterans-affairs/vets-website/actions/runs/9353273106/job/25743898597?pr=30012#step:15:32).627895:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
Couldn't find tsconfig.json. tsconfig-paths will be skipped
```

**Review by jzucadi — APPROVED** [review]

**Review by fmccaf1 — COMMENTED** [review]
Thanks for your work on this! Looks good

File: src/applications/mhv-landing-page/components/CardLayout.jsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ const CardLayout = ({ data }) => {
             mhvTransitionalMedicalRecordsLandingPage ? (
               <MedicalRecordsCard href={blueButtonUrl} />
             ) : (
-              <NavCard title={col.title} icon={col.icon} links={col.links} />
+              <NavCard {...col} />
```
Comment (radavis) [inline src/applications/mhv-landing-page/components/CardLayout.jsx:56]: Nice

File: src/applications/mhv-landing-page/components/NavCard.jsx [hunk 2]
```diff
@@ -4,7 +4,12 @@ import recordEvent from '~/platform/monitoring/record-event';
 
 export const externalLinkText = '(opens in new tab)';
 
-const NavCard = ({ icon = null, title, links }) => {
+const NavCard = ({
+  icon = null,
+  iconClasses = 'vads-u-margin-right--1p5',
```
Comment (radavis) [inline src/applications/mhv-landing-page/components/NavCard.jsx:9]: Is there a need to pass in props for icon classes?

**Review by radavis — APPROVED** [review]
Looks good

**Review by carlosfelix2 — APPROVED** [review]
You have one linting complain. Can we even replace that with va-icon?

File: src/applications/mhv-landing-page/components/NavCard.jsx (hunk 2 above)
Comment (dcloud) [reply src/applications/mhv-landing-page/components/NavCard.jsx:9]: The `attach_money` icon needed some visual adjustment due to it being a narrow piece of artwork. I figured passing iconClasses in was a reasonable way to adjust the one icon that needed it

**Comment by dcloud** [comment]
> You have one linting complain. Can we even replace that with va-icon?

@carlosfelix2 If you're taking about the one around `{!isExternal && <i aria-hidden="true" />}`, it's not a fontawesome icon, so I think the linter is confused. The closest icon is `navigate_next`, but I don't know if that would look similar enough visually.

**Pull request merged**